benchopt/runner.py

```diff
@@ -220,6 +220,33 @@ def run_one_solver(benchmark, dataset, objective, solver, n_repetitions,
             solver_description=inspect.cleandoc(solver.__doc__ or ""),
         )
 
+        obj_parameters = objective._parameters
+        slv_parameters = solver._parameters
+        ds_parameters = dataset._parameters
+
+        # Add a prefix to common keys between "meta", "obj_parameters",
+        # "slv_parameters", "ds_parameters" dictionaries to avoid
+        # erasing keys
+        all_keys = (meta.keys()
+                    | obj_parameters.keys()
+                    | slv_parameters.keys()
+                    | ds_parameters.keys())
+        common_keys = [key for key in all_keys
+                       if sum(key in d for d in
+                              (meta, obj_parameters,
+                               slv_parameters,
+                               ds_parameters)) >= 2]
+
+        for key in common_keys:
+            if key in obj_parameters:
+                obj_parameters["obj_" + key] = obj_parameters.pop(key)
+            if key in slv_parameters:
+                slv_parameters["solver_" + key] = slv_parameters.pop(key)
+            if key in ds_parameters:
+                ds_parameters["dataset_" + key] = ds_parameters.pop(key)
```

Comment:
Thinking back after our discussion, this solution is not working :(
If you have two solvers: solver1 with param `a` and solver2 with no param, and a dataset with param `a`, you will end up with 3 columns for param `a`, depending on the combination.

I think the best course of action is to use prefix all the time. It is a little bit ugly but it is robust and don't require too much dev compared to introspecting everything.